x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -111,9 +112,10 @@ public ShardFollowTasksExecutor(Client client,
 
     @Override
     public void validate(ShardFollowTask params, ClusterState clusterState) {
-        IndexRoutingTable routingTable = clusterState.getRoutingTable().index(params.getFollowShardId().getIndex());
-        if (routingTable.shard(params.getFollowShardId().id()).primaryShard().started() == false) {
-            throw new IllegalArgumentException("Not all copies of follow shard are started");
+        final IndexRoutingTable routingTable = clusterState.getRoutingTable().index(params.getFollowShardId().getIndex());
```

Comment:
I am not sure if we should remove this validation. Should we be able to create a follow task while some follower shards are temporarily unavailable?

Reply:
yeah, I'm not sure either what the point of this validation is at this specific point in the code (maybe @martijnvg can help with that). It forces retry behavior upstream for "put follow" and "resume follow" to be resilient.